feature: added description to enum types

simgobb opened the pull request:
Fixes #5619 

Changes proposed in this pull request:

- Added descriptions to enum types

matthiasrohmer commented on site/_includes/macros/render-type.njk:
```diff
@@ -206,8 +206,16 @@
+
   {# Note that we avoid whitespace below, and instead set margins in CSS. #}
-  <p class="code-sections__icon code-sections__icon--{{ iconType }}">{{ renderSingleType(type, node) | trim | safe }}
+  <p class="code-sections__icon code-sections__icon--{{ iconType }}">{{ renderSingleType(type, description) | trim | safe }}
```
I think that works for now 🙂

matthiasrohmer commented on site/_includes/macros/render-type.njk:
```diff
@@ -206,8 +206,16 @@
+
   {# Note that we avoid whitespace below, and instead set margins in CSS. #}
-  <p class="code-sections__icon code-sections__icon--{{ iconType }}">{{ renderSingleType(type, node) | trim | safe }}
+  <p class="code-sections__icon code-sections__icon--{{ iconType }}">{{ renderSingleType(type, description) | trim | safe }}
```
Looking at https://pr-6201-static-dot-dcc-staging.uc.r.appspot.com/docs/extensions/reference/offscreen/#type-Reason it seems as if the descriptions are markdown and could benefit from running through `| mdInline` 🙂

matthiasrohmer commented on site/_includes/macros/render-type.njk:
```diff
@@ -303,7 +310,7 @@
   {% elif type.type === 'literal' %}
     {# Literal value, e.g., 123 or '"foo"' #}
-    <span class="code-sections__value">{{ type.value | dump }}</span>
+    <span class="code-sections__value">{{ type.value }}</span> <br> <span class="type--xsmall" style='white-space: initial'>{{description}}<span>
```
Classic `'` vs `"`.

matthiasrohmer commented:
Hey @oliverdunk, can you have a look at this as you much better know what you were looking for than I do 🙂  See https://pr-6201-static-dot-dcc-staging.uc.r.appspot.com/docs/extensions/reference/offscreen/#type-Reason for an example. I already noted above that the descriptions' markdown would need to be rendered, but otherwise this seems what you've been looking for. Thanks!

oliverdunk commented:
This is amazing, thank you so much 🙌 

One thing we noticed is that there's maybe some other code: https://github.com/GoogleChrome/developer.chrome.com/blob/34c282f01f31f448c93893aec1755398d645767d/site/_includes/macros/render-type.njk#L275

Do you know what styles that generates? Are they nicer styles we could reuse?

Alternatively, if we could do a quick fix to remove these commas that would at least increase the information density and then I think we're good to go 😄 

<img width="399" alt="Screenshot 2023-05-04 at 18 37 30" src="https://user-images.githubusercontent.com/6097064/236284498-c72ad3bc-a558-42ce-9f18-b80dbedd0751.png">

simgobb commented:
@oliverdunk @matthiasrohmer 

I've tried the `code-sections__enum` style, but it didn't change much (the definition was black instead of blue) and it might require a slight refactor on how the macros are nested.

I removed the comma and added a new line to the original PR fix. Hope is more along the lines of what you had in mind.

oliverdunk commented:
@simgobb, thanks for checking! In that case, one more nit, could we reduce the gap a bit? e.g:

![Screenshot 2023-05-05 at 11 01 48](https://user-images.githubusercontent.com/6097064/236429555-9f2f5268-676d-4511-ae1b-8150bd08e8d0.png)

simgobb commented:
@oliverdunk 

No problem. Added a 1rem margin instead of the original <br>.

Let me know if it looks better, otherwise, I can play with that space a bit until it's right!

oliverdunk commented:
@simgobb I think enums are looking great! A couple more things:

1:

![Screenshot 2023-05-05 at 12 10 52](https://user-images.githubusercontent.com/6097064/236442962-f6a68d83-1bce-4c7a-ba36-84ef36b34053.png)

It looks like the class we changed is also used in some other places, e.g this type at https://pr-6201-static-dot-dcc-staging.uc.r.appspot.com/docs/extensions/reference/certificateProvider#type-CertificatesUpdateRequest. Is there a way we can only change enums and avoid adding extra margin to anything else?

2:

I think it probably makes sense to drop the "or" before the last item?

3:
![Screenshot 2023-05-05 at 12 17 59](https://user-images.githubusercontent.com/6097064/236444152-6dfcc385-c95a-440f-9119-ff98917e4d04.png)

It looks like for enums with more than one underscore **and** a description we accidentally include some of the enum name in the description: https://pr-6201-static-dot-dcc-staging.uc.r.appspot.com/docs/extensions/reference/certificateProvider#type-Algorithm

I think that's probably a bug here where the assumption `raw.length + 1` will be the right amount to remove doesn't hold: https://github.com/GoogleChrome/developer.chrome.com/blob/main/external/build/lib/dts-parse.js#LL799C32-L799C32

I'm not sure how familiar with that code you are, let me know if you're comfortable taking a look or if not I can probably commit something.

jpmedley commented:
@oliverdunk We're not anywhere close to finishing the review of the text that we have no. For the descriptions we have extensions engineers have been writing for at least a decade knowing that the text would NOT appear on our public site. I don't want to just turn this content on without a review.

cc: @sebastianbenz

sebastianbenz commented:
Agree with @jpmedley that we should review all descriptions. However, I would not block on this - the benefit of having descriptions outweighs  occasional spelling mistakes / errors. Review should be a quick follow up though. Created https://github.com/GoogleChromeLabs/Extension-Docs/issues/59 for tracking.

jpmedley commented:
It's not spelling mistakes I'm worried about.
Joe Medley | Technical Writer, Chrome DevRel | ***@***.*** |
 816-678-7195
*If an API's not documented it doesn't exist.*


On Mon, May 8, 2023 at 12:11 PM Sebastian Benz ***@***.***>
wrote:

> Agree with @jpmedley <https://github.com/jpmedley> that we should review
> all descriptions. However, I would not block on this - the benefit of
> having descriptions outweighs occasional spelling mistakes / errors. Review
> should be a quick follow up though. Created
> GoogleChromeLabs/Extension-Docs#59
> <https://github.com/GoogleChromeLabs/Extension-Docs/issues/59> for
> tracking.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/GoogleChrome/developer.chrome.com/pull/6201#issuecomment-1538904261>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AB6S7CYJTZC26SU54DE6KT3XFFAP3ANCNFSM6AAAAAAXV3GDFA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

oliverdunk reviewed (approved):
Approving from an extension POV, thanks so much for the fast work here :)
